internal/syncx/map.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncx
+
+import "sync"
+
+// Map is a type-safe wrapper around sync.Map.
+// It is safe for concurrent use.
+// The zero value of Map is an empty map ready to use.
+type Map[K comparable, V any] struct {
+	m sync.Map
+}
+
+func (m *Map[K, V]) Load(key K) (V, bool) {
+	v, ok := m.m.Load(key)
+	if !ok {
+		var empty V
+		return empty, false
+	}
+	return v.(V), true
+}
+
+func (m *Map[K, V]) Store(key K, value V) {
+	m.m.Store(key, value)
+}
+
+func (m *Map[K, V]) Delete(key K) {
+	m.m.Delete(key)
+}
+
+func (m *Map[K, V]) Range(f func(key K, value V) bool) {
+	m.m.Range(func(key, value any) bool {
+		return f(key.(K), value.(V))
+	})
+}
+
+func (m *Map[K, V]) LoadOrStore(key K, value V) (V, bool) {
+	v, loaded := m.m.LoadOrStore(key, value)
+	return v.(V), loaded
+}
+
+func (m *Map[K, V]) LoadAndDelete(key K) (V, bool) {
+	v, ok := m.m.LoadAndDelete(key)
+	if !ok {
+		var empty V
+		return empty, false
+	}
+	return v.(V), true
+}
+
+func (m *Map[K, V]) CompareAndSwap(key K, old, new V) bool {
+	return m.m.CompareAndSwap(key, old, new)
+}
+
+func (m *Map[K, V]) CompareAndDelete(key K, value V) bool {
+	return m.m.CompareAndDelete(key, value)
+}
+
+func (m *Map[K, V]) Swap(key K, value V) (V, bool) {
+	previous, loaded := m.m.Swap(key, value)
+	return previous.(V), loaded
+}
+
+func (m *Map[K, V]) Len() int {
```

Comment:
I believe this is not concurrent safe.

Maybe add additional field, like
```go
count atomic.Int64
```

and increment/decrement it every time we change the map?
That would also make `Len()` more performant, I think for a very low price elsewhere?